src/topology/filter.lean

```diff
@@ -0,0 +1,183 @@
+/-
+Copyright (c) 2022 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import order.filter.lift
+import topology.separation
+import data.set.intervals.monotone
+
+/-!
+# Topology on the set of filters on a type
+
+In this file introduce topology on `filter α`. It is generated by the sets
+`set.Iic (𝓟 s) = {l : filter α | s ∈ l}`, `s : set α`. A set `s : set (filter α)` is open if and
+only if it is a union of a family of these basic open sets, see `filter.is_open_iff`.
+
+This topology has the following important properties.
+
+* If `X` is a topological space, then the map `𝓝 : X → filter X` is a topology inducing map.
```

Comment:
I wonder what relationship this topology has to [Cauchy.uniform_space](https://leanprover-community.github.io/mathlib_docs/topology/uniform_space/completion.html#Cauchy.uniform_space); we have [Cauchy.dense_inducing_pure_cauchy](https://leanprover-community.github.io/mathlib_docs/topology/uniform_space/completion.html#Cauchy.dense_inducing_pure_cauchy) but that's for the pure filter, not neighborhood filter.

Reply:
I don't know. IMHO, we can look at this after merging this PR.

Reply:
Sounds good!

Reply:
Even though the relationship with [Cauchy.uniform_space](https://leanprover-community.github.io/mathlib_docs/topology/uniform_space/completion.html#Cauchy.uniform_space) is still unclear, [ultrafilter.topological_space](https://leanprover-community.github.io/mathlib_docs/topology/stone_cech.html#ultrafilter.topological_space) is just the subspace topology inherited from the topology in this PR.